src/driver-hidpp20.c

```diff
@@ -1013,21 +1021,23 @@ hidpp20drv_update_resolution_dpi(struct ratbag_resolution *resolution,
 	sensor = &drv_data->sensors[0];
 
 	/* validate that the sensor accepts the given DPI */
-	if (dpi < sensor->dpi_min || dpi > sensor->dpi_max)
-		return -EINVAL;
-	if (sensor->dpi_steps) {
-		for (i = sensor->dpi_min; i < dpi; i += sensor->dpi_steps) {
-		}
-		if (i != dpi)
+	if (!resolution->is_disabled) {
```

Comment:
Building on the comment above, I think we should move the check here.

```c
if (ratbag_resolution_has_capability(resolution, RATBAG_RESOLUTION_CAP_DISABLE) && resolution->is_disabled) {
	dpi = dpi_x = dpi_y = 0;
} else {
	if (dpi < sensor->dpi_min || dpi > sensor->dpi_max)
		...
```

Reply:
Oh yeah given the comment in the code that would make more sense.

Reply:
Ah no that doesn't work. We have to check this before since in line 1004 the dpi gets passed to `hidpp20drv_update_resolution_dpi_8100()`.